src/native/libs/System.Native/pal_time.c

```diff
@@ -58,10 +58,20 @@ uint64_t SystemNative_GetTimestamp()
 #if HAVE_CLOCK_GETTIME_NSEC_NP
     return clock_gettime_nsec_np(CLOCK_UPTIME_RAW);
 #else
+#if defined(CLOCK_BOOTTIME)
+    clockid_t clockId = CLOCK_BOOTTIME;
+#elif defined(CLOCK_MONOTONIC_RAW)
+    clockid_t clockId = CLOCK_MONOTONIC_RAW;
```

Comment:
the old solution was including suspended time, the new does not guarantee that when `CLOCK_BOOTTIME` is not defined:

https://man7.org/linux/man-pages/man2/clock_getres.2.html

> This clock does not count time that the system is suspended.

but `CLOCK_BOOTTIME` is available since Linux 2.6.39. @janvorli do you happen to know what version is being used by our official build machines? I wonder if we should worry about it.

Reply:
This is true that `CLOCK_BOOTTIME` is available on the .NET's supported Linux versions. It is also available on FreeBSD. However, on macOS and illumos, it is not available so we use the fallbacks (`CLOCK_MONOTONIC_RAW` for macOS and `CLOCK_MONOTONIC` for SunOS).

I thin the main problem with `SystemNative_GetTimestamp()` is the non-descriptive name. The name makes it sound like it will return timestamp of the current instance, and the description on its prototype does not help either:

https://github.com/dotnet/runtime/blob/f8fa9f6d1554e8db291187dd7b2847162703381e/src/native/libs/System.Native/pal_time.h#L30-L33

But in reality, before and after this PR:

* on Linux it returns the system "uptime"
* on macOS, with `CLOCK_MONOTONIC_RAW`, it returns value 4 years in the future

so it is essentially some random chronological value which we are expecting for other usages of this API.

For the purpose of this API, `Process.Linux.cs` is calling it only on Linux to obtain the uptime.

I propose, we:

* rename `SystemNative_GetTimestamp` to `SystemNative_GetArbitraryHighResolutionTimestamp` based on this excerpt from macOS manpage:
    ```
    CLOCK_MONOTONIC_RAW
    clock that increments monotonically, tracking the time since an arbitrary point
    like CLOCK_MONOTONIC.  However, this clock is unaffected by frequency or time
    adjustments.  It should not be compared to other system time sources.
    ```
* add another shim API `SystemNative_GetBootTime()`, which returns `CLOCK_BOOTTIME` on Linux and `-1` where `SystemNative_GetBootTime()` is unavailable (with a debug assert such as, `assert("SystemNative_GetBootTime is unimplemented on this platform")`).

Reply:
My personal preference would be to keep `SystemNative_GetTimestamp` unchanged (it's used by other parts of the runtime like `Stopwatch`), introduce `SystemNative_GetBootTime` which returns `-1` if `CLOCK_BOOTTIME` is not defined and fallback to the old implementation in such cases (so suspension time is always included).

The question is whether it's going to solve #67583. Is `CLOCK_BOOTTIME` defined for Android?

cc @jkotas

Reply:
> My personal preference would be to keep SystemNative_GetTimestamp unchanged

+1. The name matches the managed API that it is primarily used to implement. 

> introduce SystemNative_GetBootTime which returns -1 if CLOCK_BOOTTIME

+1 

> I don't think we still need a fallback to procfs. 

+1. We do not care about RHEL 6 or running on Linux kernels that are this old.